lib/hmr-plugin.js

```diff
@@ -8,130 +8,35 @@
-      const chunkTest = `"${chunkPlaceholder}"`;
-      const mainTest = `"${mainPlaceholder}"`;
-
-      hook.fn = (path, data) => {
-        // html-webpack-plugin doesn't call the same hooks that normal assets do. currentChunk will
-        // always be undefined for its files.
-        if (!currentChunk) {
-          return ogFn(path, data);
-        }
-
-        const { hotUpdateChunkFilename, hotUpdateMainFilename } = compiler.options.output;
-
-        if (path === chunkTest) {
-          path = `"${compiler.wpsId}-${currentChunk.id}-${chunkPlaceholder}"`;
-          compiler.options.output = Object.assign(compiler.options.output, {
-            hotUpdateChunkFilename: path.slice(1, -1),
-            oldHotUpdateChunkFilename: hotUpdateChunkFilename
-          });
-        } else if (path === mainTest) {
-          path = `"${compiler.wpsId}-${currentChunk.id}-${mainPlaceholder}"`;
-          compiler.options.output = Object.assign(compiler.options.output, {
-            hotUpdateMainFilename: path.slice(1, -1),
-            oldHotUpdateMainFilename: hotUpdateMainFilename
-          });
-        }
-
-        return ogFn(path, data);
-      };
-    });
-
-    intercept(additionalChunkAssets, 'HotModuleReplacementPlugin', (hook, ogFn) => {
-      hook.fn = (...args) => {
-        const result = ogFn(...args);
-
-        const { assets } = compilation;
-        const {
-          hotUpdateChunkFilename,
-          hotUpdateMainFilename,
-          oldHotUpdateChunkFilename,
-          oldHotUpdateMainFilename
-        } = compiler.options.output;
-        const chunkPattern = makePattern(oldHotUpdateChunkFilename);
-        const mainPattern = makePattern(oldHotUpdateMainFilename);
-
-        for (const assetName of Object.keys(assets)) {
-          if (chunkPattern.test(assetName)) {
-            assets[hotUpdateChunkFilename] = assets[assetName];
-            delete assets[assetName];
-          } else if (mainPattern.test(assetName)) {
-            assets[hotUpdateMainFilename] = assets[assetName];
-            delete assets[assetName];
-          }
-        }
-
-        return result;
-      };
-    });
+const init = function init(compiler, log) {
+  // eslint-disable-next-line no-param-reassign
+  compiler.options.output = Object.assign(compiler.options.output, {
+    hotUpdateChunkFilename: `${compiler.wpsId}-[id]-wps-hmr.js`,
+    hotUpdateMainFilename: `${compiler.wpsId}-wps-hmr.json`
   });
-};
 
-const init = function init(compiler, log) {
   const hasHMRPlugin = compiler.options.plugins.some(
     (plugin) => plugin instanceof HotModuleReplacementPlugin
   );
   if (!hasHMRPlugin) {
     addPlugin(compiler);
   } else {
-    log.warn(
+    log.error(
       'webpack-plugin-serve adds HotModuleReplacementPlugin automatically. Please remove it from your config.'
     );
+    throw new PluginExistsError(
+      'HotModuleReplacementPlugin exists in the specified configuration.'
```

Comment:
maybe provide a guide to solve this

like checking a FAQ, or saying that user should remove HotModuleReplacementPlugin from their webpack config